DMX.Core.Api.Tests.Unit/Services/Foundations/Artifacts/ArtifactServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,104 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Azure;
+using DMX.Core.Api.Models.Foundations.Artifacts;
+using DMX.Core.Api.Models.Foundations.Artifacts.Exceptions;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Foundations.Artifacts
+{
+    public partial class ArtifactServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnAddIfDependencyErrorOccursAndLogItAsync()

```

Comment:
Note for @mimendel-public and I - see if there are any other exceptions that the SDK throws

Reply:
@AndrewMcClelland let's pair on this today if you have time.